eds4jinja2/adapters/local_sparql_ds.py

```diff
@@ -29,18 +30,21 @@ def __init__(self, filename):
     def __reduce_bound_triple_to_string_format(self, dict_of_bound_variables: dict):
         return {str(k): str(v) for k, v in dict_of_bound_variables.items()}
 
-    def with_query(self, sparql_query: str) -> 'RDFFileDataSource':
+    def with_query(self, sparql_query: str, substitution_variables: dict = None) -> 'RDFFileDataSource':
         """
             Set the query text and return the reference to self for chaining.
         :return:
         """
         if self.__query__ != "":
             raise Exception("The query was already set.")
 
-        self.__query__ = sparql_query
+        if substitution_variables:
```

Comment:
are we by chance missing to cover the case when no `substitution_variables` are provided?